bulbs/content/models.py

```diff
@@ -337,8 +368,119 @@ def get_serializer_class(cls):
 
+    def percolate_special_coverages(self):
+
+        """gets list of special coverages containing this content via Elasticsearch
+        Percolator (see SpecialCoverage._save_percolator)
+
+        Results are unsorted.
+
+        """
+        special_coverage_filter = {
+            "filter": {
+                "prefix": {"_id": "specialcoverage"}
+            }
+        }
+
+        results = _percolate(index=self.mapping.index,
+                             doc_type=self.mapping.doc_type,
+                             content_id=self.id,
+                             body=special_coverage_filter)
+
+        return [r["_id"] for r in results]
+
+    def percolate_sponsored_special_coverages(self, max_size=10):
+
+        """gets list of active, sponsored special coverages containing this content via
+        Elasticsearch Percolator (see SpecialCoverage._save_percolator)
+
+        Sorting:
+            1) Manually added
+            2) Most recent start date
+        """
+
+        # Elasticsearch v1.4 percolator range query does not support DateTime range queries
+        # (PercolateContext.nowInMillisImpl is not implemented). Once using
+        # v1.6+ we can instead compare "start_date/end_date" to python DateTime
+        now_epoch = datetime_to_epoch_seconds(timezone.now())
+
+        # ES v1.4 has more limited percolator capabilities than later
+        # implementations. As such, in order to get this to work, we need to
+        # sort via scoring_functions, and then manually filter out zero scores.
+        sponsored_filter = {
+            "query": {
+                "function_score": {
+                    "functions": [
+
+                        # Boost Recent Special Coverage
+                        # Base score is start time
+                        # Note: ES 1.4 sorting granularity is poor for times
+                        # within 1 hour of each other.
+                        {
+                            "field_value_factor": {
+                                "field": "start_date",
+                            }
+                        },
+
+                        # Boost Manually Added Content
+                        {
+                            "filter": {
+                                "terms": {
+                                    "included_ids": [self.id],
+                                }
+                            },
+                            "weight": 10,
+                        },
+                        # Penalize Unsponsored
+                        {
+                            "filter": {
+                                "term": {
+                                    "sponsored": False,
+                                }
+                            },
+                            "weight": 0,
+                        },
+                        # Penalize Inactive
+                        {
+                            "filter": {
+                                "or": [
```

Comment:
Maybe it's lower in the PR and I'll see it, but I wanted to make sure we're completely excluding inactive special coverage from consideration and not just penalizing